Database branch

benfrey opened the pull request:
Will have the expressAPIServer and expressWebServerExample directories working properly (the same as the expressWebServerExampleStandalone) by tomorrow night.

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Create classy-schedule-network-layout.png

benfrey committed:
Add files via upload

benfrey committed:
Update expressAPIServer

benfrey committed:
Update documentation for each directory

benfrey committed:
Update README.md

benfrey committed:
Update README.md

gabbiebolcer commented on expressAPIServer/app.js:
```diff
@@ -1,48 +1,43 @@
+// Eventually we would like to replace this authentificaiton with tokens
+var config =
+{
+    host: 'classy-schedule-database.mysql.database.azure.com';,
+    user: 'db_test';,
+    password: 'fA!6#_&eaU9-EaeJ';,
+    database: 'db_dev';,
```
should there be `;` here? when I copied this into my IDE it was giving me errors

benfrey committed:
Work in progress

Debating whether the helper function should just create an HTTP path or if they should actually submit the HTTP request

benfrey committed:
Still an issue with row population

benfrey committed:
Update API readme

benfrey committed:
Delete expressAPIServer/.vscode directory

benfrey committed:
Delete expressAPIServer/routes directory

benfrey committed:
Delete expressAPIServer/views directory

benfrey committed:
Delete expressAPIServer/models directory

benfrey committed:
Delete expressAPIServer/public/stylesheets directory

benfrey committed:
Delete .deployment

benfrey committed:
Update Readme

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Trying to deploy

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Issue with deployment but done for tonight

benfrey committed:
Merge branch 'Database-Branch' of https://github.com/benfrey/CISC480-Spring-2022 into Database-Branch

benfrey committed:
Update README.md

benfrey committed:
Update README.md

benfrey committed:
Update README.md

joelambrecht commented on expressAPIServer/README.md:
```diff
@@ -0,0 +1,45 @@
+- **View something** (i.e. GET): $curl -X GET "http://localhost:4000/getDept"
+- **View something** (i.e. GET): $curl -X GET "http://localhost:4000/getClass?dept_code=STAT"
+- **Add something** (i.e. POST): $curl -X POST "http://localhost:4000/addClass?dept_code=STAT&class_num=101&class_name=Introduction+to+Statistics"
+    - Note: If an error from database is the response, the entry probably already exists or something more serious happened. Either way, we are working to increase error verbosity.
```
Not necessarily increase verbosity- don't want to decrease security.

joelambrecht commented on expressAPIServer/README.md:
```diff
@@ -0,0 +1,45 @@
+From another terminal, submit HTTP requests to "http://classy-api.ddns.net". Here are some examples to connect to the API via [cURL](https://curl.se/):
+- **View something** (i.e. GET): $curl -X GET "http://classy-api.ddns.net/getDept"
+- **View something** (i.e. GET): $curl -X GET "http://classy-api.ddns.net/getClass?dept_code=STAT"
+- **Add something** (i.e. POST): $curl -X POST "http://classy-api.ddns.net/addClass?dept_code=STAT&class_num=101&class_name=Introduction+to+Statistics"
```
add uses [data], so this should be $curl -X POST -d "dept_code=STAT&class_num=220&class_name=Statistics 1" ""http://classy-api.ddns.net/addClass"

joelambrecht replied:
curl also defaults to get, so you don't need the -X for those commands

joelambrecht commented on expressAPIServer/app.js:
```diff
@@ -1,115 +1,92 @@
-});
+}
 
 module.exports = app;
```
We'll need to compare my API file to yours here- hard to tell what changes are mandatory and which are aesthetic.

joelambrecht reviewed (commented):
As this gets closer to being shared with other teams, keep in mind what schema you have it pointing to. db_dev is our team's development environment, but we want to direct other teams to cs_dev.

benfrey committed:
Update README.md

benfrey committed:
Update .gitignore

benfrey committed:
Update app.js
